storage/storage_model.go

```diff
@@ -103,3 +119,55 @@ func (m *buildMetadata) toJSON() ([]byte, error) {
 
 	return metadataJSON, nil
 }
+
+func (m *buildMetadata) fromJSON(data []byte) error {
+	return errors.Wrap(json.Unmarshal(data, m), "unmarshalling metadata")
+}
+
+type testMetadata struct {
+	ID      string `json:"id"`
+	Name    string `json:"name"`
+	BuildID string `json:"build_id"`
+	TaskID  string `json:"task_id"`
+}
+
+func newTestMetadata(t model.Test) testMetadata {
+	return testMetadata{
+		ID:      t.Id.String(),
```

Comment:
I think we want `t.Id.Hex()` since the String representation adds some extra cruft.